spec/index.bs

```diff
@@ -1324,7 +1330,98 @@ dictionary GPUBufferBinding {
+    ::
+        The set of {{GPUBindGroupBinding}}s this {{GPUBindGroup}} describes.
+</dl>
+
+### {{GPUDevice/createBindGroup()|GPUDevice.createBindGroup(GPUBindGroupDescriptor)}} ### {#GPUDevice-createBindGroup}
+
+<div algorithm="GPUDevice.createBindGroup(descriptor)">
+
+The <dfn method for="GPUDevice">createBindGroup(|descriptor|)</dfn> method is used to create
+    {{GPUBindGroup}}s.
+
+1. Ensure [=bind group device validation=] is not violated.
+1. Ensure |descriptor|.{{GPUBindGroupDescriptor/layout}} is a valid {{GPUBindGroupLayout}}.
+1. Ensure the number of {{GPUBindGroupLayoutDescriptor/bindings}} of
+    |descriptor|.{{GPUBindGroupDescriptor/layout}} exactly equals to the number of
+    |descriptor|.{{GPUBindGroupDescriptor/bindings}}.
+1. For each {{GPUBindGroupBinding}} |bindingDescriptor| in
+    |descriptor|.{{GPUBindGroupDescriptor/bindings}}:
+    1. Ensure there is exactly one {{GPUBindGroupLayoutBinding}} |layoutBinding| in
+        {{GPUBindGroupLayoutDescriptor/bindings}} of |descriptor|.{{GPUBindGroupDescriptor/layout}}
+        such that |layoutBinding|.{{GPUBindGroupLayoutBinding/binding}} equals to
+        |bindingDescriptor|.{{GPUBindGroupBinding/binding}}.
+    1. If |layoutBinding|.{{GPUBindGroupLayoutBinding/type}} is
+        {{GPUBindingType/"sampler"}}:
+        1. Ensure |bindingDescriptor|.{{GPUBindGroupBinding/resource}} is a
+            valid {{GPUSampler}} object.
+    1. If |layoutBinding|.{{GPUBindGroupLayoutBinding/type}} is
+        {{GPUBindingType/"sampled-texture"}} or {{GPUBindingType/"storage-texture"}}.
+        1. Ensure |bindingDescriptor|.{{GPUBindGroupBinding/resource}} is a
+            valid {{GPUTextureView}} object.
+        1. Ensure [=texture view binding validation=] is not violated.
+    1. If |layoutBinding|.{{GPUBindGroupLayoutBinding/type}} is
+        {{GPUBindingType/"uniform-buffer"}} or {{GPUBindingType/"storage-buffer"}}
+        or {{GPUBindingType/"readonly-storage-buffer"}}.
+        1. Ensure |bindingDescriptor|.{{GPUBindGroupBinding/resource}} is a
+            valid {{GPUBufferBinding}} object.
+        1. Ensure [=buffer binding validation=] is not violated.
+1. Return a new {{GPUBindGroup}} object with:
+    - {{GPUBindGroup/[[layout]]}} = |descriptor|.{{GPUBindGroupDescriptor/layout}}
+    - {{GPUBindGroup/[[bindings]]}} = |descriptor|.{{GPUBindGroupDescriptor/bindings}}
+
+<b>Validation Conditions</b>
+
+<dl dfn-for="createBindGroup(descriptor)">
+If any of the following conditions are violated:
+    1. Generate a {{GPUValidationError}} in the current scope with appropriate error message.
+    1. Create a new invalid {{GPUBindGroup}} and return the result.
+
+<dfn>bind group device validation</dfn>: The {{GPUDevice}} must not be lost.
+
+<dfn>texture view binding validation</dfn>: A compatible texture BindGroupLayout binding must exist
+    in {{GPUBindGroupLayoutDescriptor/bindings}} of |descriptor|.{{GPUBindGroupDescriptor/layout}}.
+    This requires:
```

Comment:
Should specify *where* in the bindings. E.g.
```
<dfn>texture view binding validation</dfn>:
Let |view| be |bindingDescriptor|.{{GPUBindGroupBinding/resource}}, a {{GPUTextureView}}.
This |layoutBinding| must be compatible with this |view|. This requires:
```

Reply:
Good suggestion. It's more formal now. Done!